### PR TITLE
Moveit_cpp Tutorial

### PR DESCRIPTION
**Pull request by MarqRazz**
Port of moveit_cpp tutorial to ROS2. This addresses issue #20

To launch the tutorial run:
`ros2 launch moveit2_tutorials moveit_cpp_tutorial.launch.py

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Convert launch files to ROS2
- [x] Cleanup demo and add documentation

### CODE REVIEW

File: doc/moveit_cpp/config/panda_controllers.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+panda_arm_controller:
```
Comment (JafarAbdi) [inline doc/moveit_cpp/config/panda_controllers.yaml:1]: This's for the old ros2_control, we're depending on the master branch now, see [run_moveit_cpp.launch.py](https://github.com/ros-planning/moveit2/blob/main/moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py)

File: doc/moveit_cpp/config/controllers.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+controller_names:
```
Comment (JafarAbdi) [inline doc/moveit_cpp/config/controllers.yaml:1]: We should use the one from [moveit_resources](https://github.com/ros-planning/moveit_resources/blob/ros2/panda_moveit_config/config/panda_controllers.yaml)

File: doc/moveit_cpp/config/start_positions.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+fake_joint_driver_node:
```
Comment (JafarAbdi) [inline doc/moveit_cpp/config/start_positions.yaml:1]: This's no longer needed

File: doc/moveit_cpp/config/moveit_cpp.yaml [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-planning_pipelines:
-  pipeline_names:
-    - ompl
-
-plan_request_params:
-  planning_attempts: 1
-  planning_pipeline: ompl
-  max_velocity_scaling_factor: 1.0
-  max_acceleration_scaling_factor: 1.0
+moveit_cpp_tutorial:
```
Comment (JafarAbdi) [inline doc/moveit_cpp/config/moveit_cpp.yaml:1]: Should we move this file to moveit_resources .?

File: doc/moveit_cpp/launch/moveit_cpp_tutorial.launch.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    # Publish TF
+    robot_state_publisher = Node(package='robot_state_publisher',
+                                 executable='robot_state_publisher',
+                                 name='robot_state_publisher',
+                                 output='both',
+                                 parameters=[robot_description])
+
+    # Fake joint driver
+    fake_joint_driver_node = Node(package='fake_joint_driver',
```
Comment (JafarAbdi) [inline doc/moveit_cpp/launch/moveit_cpp_tutorial.launch.py:91]: We no longer depend on `fake_joint_driver` [see](https://github.com/ros-planning/moveit2/blob/main/moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py#L135)

**Review by JafarAbdi — CHANGES_REQUESTED** [review]
This will not work with the main branch of moveit2

**Comment by JafarAbdi** [comment]
Should we remove `run_moveit_cpp` from the MoveIt2 repo and point users to this page .?

**Comment by MarqRazz** [comment]
Changed were added in #69

**Pull request closed without merging**